presto-main/src/main/java/com/facebook/presto/type/TypeRegistry.java

```diff
@@ -458,16 +500,108 @@ else if (isCovariantParameterPosition(commonSuperTypeBase.get(), i)) {
         return Optional.of(new TypeSignature(commonSuperTypeBase.get(), typeParameters.build()));
     }
 
-    /*
-     * Given signature with unmatched parameters return filtered one.
+    /**
+     * In case of coercion between un-parametrized types (DOUBLE, BIGINT, ...)
+     * we only need to know whether coercion is allowed (true|false).
+     * But in case of coercion to the type with long parameters, parameters of the result type must be calculated.
+     * Such coercion rules must be implemented within this method.
+     * <p>
+     * e.g.:
+     * LONG -> DECIMAL = DECIMAL(19,0)
```

Comment:
BTW, I don't think this is correct, anyway. `DECIMAL` is syntactic sugar for `DECIMAL(38, 0)`, so:

```
common-supertype(BIGINT, DECIMAL) = common-supertype(BIGINT, DECIMAL(38,0)) = DECIMAL(38, 0)
```